### PR TITLE
Cleaned up exit routine code

### PR DESCRIPTION
**Pull request by ael-code**
The exit routine code is executed on every program run, thus
I consider it pretty important.
I found out that this part of the code was very confuse and complex,
thus very difficult to modify.

Moreover the message and the exit code brought up with the
`sys.exit` function or the `SystemExit` exception were completely ignored.
In fact the `return` statement into the exception handler was bypassed by
the `finally` block.

I made a little bit of refracotring and I hope that this will make easier to
mantain this part of the code.

The second commit is a simple pep8 error fix

### CODE REVIEW

**Comment by ael-code** [comment]
I've a PR  very very nice, almost ready, that will fix #713 :) but it depends on this. Let me know if you need more explanations about the code changes in order to proceeds.

I think that if you compare the new `finally` block wrt the old one you get easily convinced that these changes are very useful for the maintenance of the code

File: ranger/core/main.py [hunk 1]
```diff
@@ -86,7 +86,9 @@ def print_function(string):
             rifle.execute(targets, number=ranger.arg.mode, flags=ranger.arg.flags)
             return 1 if arg.fail_unless_cd else 0  # COMPAT
 
-    crash_traceback = None
+    profile = None
+    exit_mex = None
```
Comment (nfnty) [inline ranger/core/main.py]: Rename `exit_mex` -> `exit_msg`

Initialize as empty string: `exit_msg = ''`

File: ranger/core/main.py [hunk 2]
```diff
@@ -137,43 +139,46 @@ def print_function(string):
         if ranger.arg.profile:
             import cProfile
             import pstats
-            profile = None
             ranger.__fm = fm
-            cProfile.run('ranger.__fm.loop()', tempfile.gettempdir() + '/ranger_profile')
-            profile = pstats.Stats(tempfile.gettempdir() + '/ranger_profile', stream=sys.stderr)
+            profile_file = tempfile.gettempdir() + '/ranger_profile'
```
Comment (nfnty) [inline ranger/core/main.py]: Use `os.path.join()`

File: ranger/core/main.py [hunk 3]
```diff
@@ -137,43 +139,46 @@ def print_function(string):
...
+            profile = pstats.Stats(profile_file, stream=sys.stderr)
         else:
             fm.loop()
+
     except Exception:
         import traceback
-        crash_traceback = traceback.format_exc()
-    except SystemExit as error:
-        return error.args[0]
+        exit_mex = "ranger version: {0}, executed with python {1}\n".format(
```
Comment (nfnty) [inline ranger/core/main.py]: `exit_msg =` -> `exit_msg +=`

File: ranger/core/main.py [hunk 4]
```diff
@@ -137,43 +139,46 @@ def print_function(string):
...
+                            ranger.__version__, sys.version.split()[0])
+        exit_mex = exit_mex + "Locale: %s\n" % '.'.join(str(s) for s in locale.getlocale())
```
Comment (nfnty) [inline ranger/core/main.py]: Replace all `exit_msg = exit_msg +` -> `exit_msg +=`

File: ranger/core/main.py [hunk 5]
```diff
@@ -137,43 +139,46 @@ def print_function(string):
...
+        try:
+            exit_mex = exit_mex + "Current file: '{0}'\n".format(fm.thisfile.path)
+        except Exception:
+            pass
+        exit_mex = exit_mex + "\n" + traceback.format_exc() + "\n"
+        exit_mex = exit_mex + ("ranger crashed. Please report this traceback at:\n"
+                              "https://github.com/hut/ranger/issues")
+        exit_code = 1
+
+    except SystemExit as ex:
```
Comment (nfnty) [inline ranger/core/main.py]: Rename `ex` -> `error`

File: ranger/core/main.py [hunk 6]
```diff
@@ -137,43 +139,46 @@ def print_function(string):
...
+        if ex.code is not None:
+            if type(ex.code) is not int:
+                exit_mex = ex.code
+                exit_code = 1
+            else:
+                exit_code = ex.code
```
Comment (nfnty) [inline ranger/core/main.py]:
```
if error.code is not None:
    if isinstance(error.code, int) and 0 <= error.code <= 127:
        exit_code = error.code
    else:
        exit_code = 1
        exit_msg += 'SystemExit: {0}'.format(error.code)
```

File: ranger/core/main.py [hunk 7]
```diff
@@ -137,43 +139,46 @@ def print_function(string):
...
+        exit_mex = "ranger version: {0}, executed with python {1}\n".format(
+                            ranger.__version__, sys.version.split()[0])
+        exit_mex = exit_mex + "Locale: %s\n" % '.'.join(str(s) for s in locale.getlocale())
+        try:
+            exit_mex = exit_mex + "Current file: '{0}'\n".format(fm.thisfile.path)
+        except Exception:
+            pass
+        exit_mex = exit_mex + "\n" + traceback.format_exc() + "\n"
+        exit_mex = exit_mex + ("ranger crashed. Please report this traceback at:\n"
+                              "https://github.com/hut/ranger/issues")
```
Comment (nfnty) [inline ranger/core/main.py]: Add `\n`

File: ranger/core/main.py [hunk 8]
```diff
@@ -137,43 +139,46 @@ def print_function(string):
...
-                pass
-            print(crash_traceback)
-            print("ranger crashed.  "
-                "Please report this traceback at:")
-            print("https://github.com/hut/ranger/issues")
-            return 1
-        return 0
+        # print the exit message if any
+        if exit_mex:
+            sys.stderr.write(exit_mex + '\n')
```
Comment (nfnty) [inline ranger/core/main.py]: `print(exit_msg, file=sys.stderr)`

**Comment by nfnty** [comment]
Rebase this PR and make sure that all tests pass. See the updated `CONTRIBUTING` and `HACKING` documents for more info. Thanks!

**Pull request merged**